### PR TITLE
Add task for swap demo

### PR DESCRIPTION
**Pull request by Rolaman**
## Description
Add hardhat task to show minting, swapping and burning via the pair contract

## What type of PR is this? (check all applicable)
- [x] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Test

## Steps to QA
No need

### CODE REVIEW

File: tasks/core/demo.ts [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+    const reserves = await pair.getReserves();
+    console.log(
+      "BURN RESULT: Reserves from pair after burn:",
+      reserves[0].toString(),
+      reserves[1].toString(),
+    );
+  },
+);
+
+export async function deployNilContract(
```
Comment (0xAleksaOpacic) [inline tasks/core/demo.ts:347]: This should go into some utility file

File: tasks/core/demo.ts [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+
+  if (!txReceipt || !txReceipt.contractAddress) {
+    throw new Error("Contract deployment failed");
+  }
+
+  const deployedContract = factory.attach(txReceipt.contractAddress);
+  return { deployedContract, contractAddress: txReceipt.contractAddress };
+}
+
+function calculateOutputAmount(
```
Comment (0xAleksaOpacic) [inline tasks/core/demo.ts:368]: This as well, also make sire swap is also calling it

File: tasks/router/router_1.ts [hunk 3]
```diff
@@ -9,7 +9,7 @@ import {
 import { task } from "hardhat/config";
 import { encodeFunctionData } from "viem";
```
Comment (0xAleksaOpacic) [inline tasks/router/router_1.ts:10]: What is this file?

File: tasks/core/demo.ts (hunk 1 above)
Comment (Rolaman) [reply tasks/core/demo.ts]: done

File: tasks/router/router_1.ts (hunk 3 above)
Comment (Rolaman) [reply tasks/router/router_1.ts]: draft file to call router contract, removed

**Pull request merged**